daemons/fenced/fenced_commands.c

```diff
@@ -2104,6 +2104,13 @@ can_fence_host_with_device(stonith_device_t *dev,
         goto search_report_results;
     }
 
+    // Short-circuit if device does not support action
+    if (pcmk__str_eq(action, "on", pcmk__str_none)
+        && !pcmk_is_set(dev->flags, st_device_supports_on)) {
+        check_type = "Agent does not support 'on'";
+        goto search_report_results;
+    }
+
```

Comment:
We could combine that to
/* Short-circuit query if device doesn't allow action or
    this host is not allowed to perform the action */
removing a compare.
If it improves/simplifies readability probably depends on personal preference.

Reply:
makes sense